GitCommands/ArgumentBuilderExtensions.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel;
+using GitCommands.Git;
+using JetBrains.Annotations;
+
+namespace GitCommands
+{
+    /// <summary>
+    /// Extension methods for working with <see cref="ArgumentBuilder"/>.
+    /// </summary>
+    public static class ArgumentBuilderExtensions
+    {
+        /// <summary>
+        /// Adds <paramref name="ifConditionTrue"/> to the argument list if <paramref name="condition"/>
+        /// is <c>true</c>.
+        /// </summary>
+        /// <remarks>
+        /// If <paramref name="ifConditionTrue"/> is <c>null</c> or white-space, then no change is made
+        /// to the argument list, regardless of the value of <paramref name="condition"/>.
+        /// </remarks>
+        /// <param name="builder">The <see cref="ArgumentBuilder"/> to add arguments to.</param>
+        /// <param name="condition">Whether or not to add <paramref name="ifConditionTrue"/> to the argument list.</param>
+        /// <param name="ifConditionTrue">The string to add if <paramref name="condition"/> is <c>true</c>.</param>
+        public static void Add(this ArgumentBuilder builder, bool condition, [CanBeNull] string ifConditionTrue)
```

Comment:
:bulb: Consider making the multiple-argument forms take a tuple type instead of simply multiple arguments. This will allow use of named arguments at the call site with compiler validation.

```
public static void Add(this ArgumentBuilder builder, (bool condition, string ifConditionTrue) value)
```

```
new ArgumentBuilder
{
  (condition: true, ifConditionFalse: "") // Compiler error
}
```

Reply:
Could you please explain how the above causes a compiler warning

Reply:
Because of the label mismatch. `ifConditionTrue` != `ifConditionFalse`.